### ISSUE TITLE
Variable Types

### ISSUE THREAD

**Issue by JimmyRooster** [GrooveProgramming/editor#1]
@B294 @Spaceflyer234 

### Overview
Scratch has two different variable types: bools (hexagon shape) and strings / ints (rounded). These types can't be mixed. I think we should allow users to make variables of type bool and keep bools separate from strings / ints. Do you agree?

I was also thinking we could have variables that are a reference to a sprite/clone if we had the idea where you can call a built in / custom function on another sprite. This would allow you to have a block that takes in a variable of this type and a block (like the execute function in snap).

Should there be any other types of variables?

**Comment by B294** [comment]
We can make cloud variables once we host the database and server, I think we should use DynamoDB. Then we can stay here as well. Is this a feature? At this point of time, when we haven’t finished the stage or paint editor, is it really that necessary? I think it’s a good idea though. I don’t know how these would look UI wise.

**Comment by Spaceflyer234** [comment]
We need to fix the editor...

**Comment by JimmyRooster** [comment]
@B294  I wasn't talking specifically about cloud variables, just normal variables. They are integral to the editor as most blocks can take in a variable.

**Comment by JimmyRooster** [comment]
@Spaceflyer234 What a good idea, I'll try and do it now.

**Comment by B294** [comment]
> @Spaceflyer234 What a good idea, I'll try and do it now.

Lol

**Comment by B294** [comment]
> @B294 I wasn't talking specifically about cloud variables, just normal variables. They are integral to the editor as most blocks can take in a variable.

Alright, I’m on the chat server. https://jemunm378.github.io/ScratchHelp.github.io/Chat.html

**Comment by Spaceflyer234** [comment]
Once we get accounts and stuff working we should make a groove team account.

**Comment by JimmyRooster** [comment]
> > @Spaceflyer234 What a good idea, I'll try and do it now.
> 
> Lol

Mission complete.

**Comment by Spaceflyer234** [comment]
Alright, who stole the run code button?

**Comment by JimmyRooster** [comment]
> Alright, who stole the run code button?

I did...
I commented that out because the run code button is not on the code editor page but on the stage

**Comment by Spaceflyer234** [comment]
Okay.

**Comment by B294** [comment]
> Alright, who stole the run code button?

Well, it’s on the stage. :P

**Comment by B294** [comment]
> Okay.

Meet me here? https://jemunm378.github.io/ScratchHelp.github.io/Chat.html

**Comment by Spaceflyer234** [comment]
> > Alright, who stole the run code button?
> 
> I did...
> I commented that out because the run code button is not on the code editor page but on the stage

I guess you can just type runCode() in the JS console to run the code.

**Comment by B294** [comment]
> > > Alright, who stole the run code button?
> > 
> > 
> > I did...
> > I commented that out because the run code button is not on the code editor page but on the stage
> 
> I guess you can just type runCode() in the JS console to run the code.

Will you meet me here to work on Groove?
https://jemunm378.github.io/ScratchHelp.github.io/Chat.html

**Comment by Spaceflyer234** [comment]
We could use https://github.com/discourse/discourse as a forum.

**Comment by B294** [comment]
> We could use https://github.com/discourse/discourse as a forum.

Hmm, I’ll check it out.

**Comment by Spaceflyer234** [comment]
Morphic.js could help us a big deal too.

**Comment by B294** [comment]
> Morphic.js could help us a big deal too.

What?

**Comment by Spaceflyer234** [comment]
> > Morphic.js could help us a big deal too.
> 
> What?

https://github.com/jmoenig/morphic.js/blob/master/morphic.js

**Comment by B294** [comment]
> > > Morphic.js could help us a big deal too.
> > 
> > 
> > What?
> 
> https://github.com/jmoenig/morphic.js/blob/master/morphic.js

Oh, can you meet me on the chat server? https://jemunm378.github.io/ScratchHelp.github.io/Chat.html

**Comment by B294** [comment]
So we can talk about Groove there.

**Comment by Spaceflyer234** [comment]
I have added save/load functions that can be used by running the command `saveCode()` and `loadCode() ` in the JS console.

**Comment by B294** [comment]
> I have added save/load functions that can be used by running the command `saveCode()` and `loadCode() ` in the JS console.

Wait... You did that without a server?

**Comment by Spaceflyer234** [comment]
> > I have added save/load functions that can be used by running the command `saveCode()` and `loadCode() ` in the JS console.
> 
> Wait... You did that without a server?

Yes, also do you like how it creates a "my variable" at the start?

**Comment by B294** [comment]
> > > I have added save/load functions that can be used by running the command `saveCode()` and `loadCode() ` in the JS console.
> > 
> > 
> > Wait... You did that without a server?
> 
> Yes, also do you like how it creates a "my variable" at the start?

It’s like Scratch, it’s an ok feature, especially for those who don’t know how to make one.

**Comment by B294** [comment]
> > > > Morphic.js could help us a big deal too.
> > > 
> > > 
> > > What?
> > 
> > 
> > https://github.com/jmoenig/morphic.js/blob/master/morphic.js
> 
> Oh, can you meet me on the chat server? https://jemunm378.github.io/ScratchHelp.github.io/Chat.html

Don’t meet me on now, i have to go. But should we chat here?
